### PR TITLE
Improve networking

### PR DESCRIPTION
**Pull request by kron4eg**
**What this PR does / why we need it**:

* Upgrade Canal to v3.15.1
* Make ubuntu 20.04 work unconditionally
* Canal CNI got new MTU field
* Default Canal MTU, taking into account different providers
* Label everything we deploy with "kubeone.io/component: <COMPONENT-NAME>"

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
* Upgrade Canal to v3.15.1
* Make ubuntu 20.04 work unconditionally
* Canal CNI got new MTU field
* Default Canal MTU, taking into account different providers
* Label everything we deploy with "kubeone.io/component: <COMPONENT-NAME>"
```

### CODE REVIEW

File: pkg/apis/kubeone/v1beta1/defaults.go [hunk 1]
```diff
@@ -34,6 +34,8 @@ const (
 	DefaultNodePortRange = "30000-32767"
 	// DefaultStaticNoProxy defined static NoProxy
 	DefaultStaticNoProxy = "127.0.0.1/8,localhost"
+	// Default MTU
+	DefaultMTU = 1500
```
Comment (xmudrii) [inline pkg/apis/kubeone/v1beta1/defaults.go:38]: I don't see that this variable is used anywhere.

File: pkg/apis/kubeone/v1beta1/defaults.go [hunk 2]
```diff
@@ -127,8 +129,23 @@ func SetDefaults_ClusterNetwork(obj *KubeOneCluster) {
...
+		switch {
+		case obj.CloudProvider.AWS != nil:
+			canal.MTU = "8951" // 9001 AWS Jumbo Frame - 50 VXLAN bytes
+		case obj.CloudProvider.GCE != nil:
+			canal.MTU = "1410" // GCE specific 1460 bytes - 50 VXLAN bytes
+		case obj.CloudProvider.Hetzner != nil:
+			canal.MTU = "1400" // Hetzner specific 1450 bytes - 50 VXLAN bytes
+		case obj.CloudProvider.Openstack != nil:
+			canal.MTU = "1400" // Openstack specific 1450 bytes - 50 VXLAN bytes
+		}
```
Comment (xmudrii) [inline pkg/apis/kubeone/v1beta1/defaults.go:145]: MTU should be defaulted regardless of is `clusterNetwork.CNI == nil`. We should skip defaulting only if `MTU != ""`.

File: pkg/apis/kubeone/types.go [hunk 3]
```diff
@@ -182,7 +182,9 @@ type CNI struct {
 }
 
 // CanalSpec defines the Canal CNI plugin
-type CanalSpec struct{}
+type CanalSpec struct {
+	MTU string `json:"mtu"`
```
Comment (xmudrii) [inline pkg/apis/kubeone/types.go:186]: Why is MTU a string and not an int?

File: pkg/clientutil/create_or_update.go [hunk 4]
```diff
@@ -28,8 +28,20 @@ import (
...
+	return func(c client.Client, obj runtime.Object) {
+		LabelComponent(componentname, obj)
+	}
+}
+
 // CreateOrUpdate makes it easy to "apply" objects to kubernetes API server
-func CreateOrUpdate(ctx context.Context, c client.Client, obj runtime.Object) error {
+func CreateOrUpdate(ctx context.Context, c client.Client, obj runtime.Object, updaters ...Updater) error {
+	for _, update := range updaters {
+		update(c, obj)
```
Comment (xmudrii) [inline pkg/clientutil/create_or_update.go:42]: Where is this update function defined? I don't see it.

File: pkg/apis/kubeone/types.go (hunk 3 above)
Comment (kron4eg) [reply pkg/apis/kubeone/types.go:186]: Because it's being consumed as string

File: pkg/clientutil/create_or_update.go (hunk 4 above)
Comment (kron4eg) [reply pkg/clientutil/create_or_update.go:42]: right here, `WithComponentLabel()`

File: pkg/apis/kubeone/v1beta1/defaults.go (hunk 2 above)
Comment (kron4eg) [reply pkg/apis/kubeone/v1beta1/defaults.go:145]: If I'll assign to obj.ClusterNetwork.CNI regardless of its nilness, that will override other possible options selected by the user (weave / external), isn't it?

**Comment by kron4eg** [comment]
@xmudrii comments addressed

**Pull request merged**